examples/Example.elm

```diff
@@ -0,0 +1,149 @@
+
+
+type alias Todo =
+    { id : String
+    , title : Maybe String
+    , description : Maybe String
+    , last_modified : Int
+    }
+
+
+
+-- And how to decode from a Kinto record to a Todo...
+
+
+decodeTodo : Decode.Decoder Todo
+decodeTodo =
+    (Decode.map4 Todo
+        (Decode.field "id" Decode.string)
+        (Decode.maybe (Decode.field "title" Decode.string))
+        (Decode.maybe (Decode.field "description" Decode.string))
+        (Decode.field "last_modified" Decode.int)
+    )
+
+
+
+-- And how to encode Todo data to send to Kinto.
+
+
+encodeData : Maybe String -> Maybe String -> Encode.Value
+encodeData title description =
+    Encode.object
+        [ ( "title", Encode.string (Maybe.withDefault "" title) )
+        , ( "description", Encode.string (Maybe.withDefault "" description) )
+        ]
+
+
+
+-- Configure a Kinto client: the server url and authentication
+
+
+client : Kinto.Client
+client =
+    Kinto.client
+        "https://kinto.dev.mozaws.net/v1/"
+        (Kinto.Basic "test" "test")
+
+
+
+-- Set up the resource we're interested in: in this case, Kinto records, stored
+-- in the "test-items" collection, in the "default" bucket.
+-- We give it the `decodeTodo` decoder so the data we get back from the Kinto
+-- server will be properly decoded into our `Todo` record.
+
+
+recordResource : Kinto.Resource Todo
+recordResource =
+    Kinto.recordResource "default" "test-items" decodeTodo
+
+
+
+-- Add a new Todo
+
+
+addTodo : Maybe String -> Maybe String -> Cmd Msg
+addTodo title description =
+    let
+        data =
+            encodeData title description
+    in
+        client
+            |> Kinto.create recordResource data
+            |> Kinto.send TodoAdded
+
+
+
+-- Get all the Todos
+
+
+getTodos : Cmd Msg
+getTodos =
+    client
+        |> Kinto.getList recordResource
+        |> Kinto.sortBy [ "title", "description" ]
+        |> Kinto.send TodosFetched
+
+
+
+-- Tie it all together
+
+
+type alias Model =
+    { todos : List Todo
+    , title : Maybe String
+    , description : Maybe String
+    , error : Maybe String
+    }
+
+
+init : ( Model, Cmd Msg )
+init =
+    Model [] Nothing Nothing Nothing
+        ! [ addTodo (Just "test") (Just "description")
+          , getTodos
+          ]
+
+
+type Msg
+    = TodoAdded (Result Kinto.Error Todo)
+    | TodosFetched (Result Kinto.Error (List Todo))
+
+
+view : Model -> Html.Html Msg
+view model =
+    Html.text <| toString model.todos
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update message model =
+    case message of
+        TodoAdded (Ok todo) ->
+            { model | error = Nothing } ! []
+
+        TodoAdded (Err error) ->
+            { model | error = Just <| toString error } ! []
+
+        TodosFetched (Ok todoList) ->
+            { model | error = Nothing, todos = todoList } ! []
+
+        TodosFetched (Err error) ->
+            { model | error = Just <| toString error } ! []
```

Comment:
It sure does feel like there's a lot of boilerplate here. Since we don't actually do anything different on the different error cases, we could make `TodoAdded` and `TodosFetched` only handle the `Ok` case, and define a single `CommunicationError` type to handle the shared `Error` cases.